internal/internal_worker.go

```diff
@@ -1827,8 +1839,18 @@ func getReadOnlyChannel(c chan struct{}) <-chan struct{} {
 }
 
 func setWorkerOptionsDefaults(options *WorkerOptions) {
+	if options.Tuner != nil {
+		if options.MaxConcurrentWorkflowTaskExecutionSize != 0 ||
+			options.MaxConcurrentActivityExecutionSize != 0 ||
+			options.MaxConcurrentLocalActivityExecutionSize != 0 {
+			panic("cannot set MaxConcurrentWorkflowTaskExecutionSize, MaxConcurrentActivityExecutionSize, or MaxConcurrentLocalActivityExecutionSize with Tuner")
+		}
+	}
+	maxConcurrentWFT := options.MaxConcurrentWorkflowTaskExecutionSize
+	maxConcurrentAct := options.MaxConcurrentActivityExecutionSize
+	maxConcurrentLA := options.MaxConcurrentLocalActivityExecutionSize
```

Comment:
Can probably just mutate the options still at this point since I don't expect these three fields are ever even referenced anymore after this function returns, but no big deal of course

Reply:
That actually causes problems since sometimes the tests will end up passing these around and this function gets run more than once and then the checks that you haven't set both maxConcurrentXXX and Tuner fail.